### ISSUE TITLE
Static checks do not run on non-x86

### ISSUE THREAD

**Issue by Amulyam24** [kata-containers/tests#4026]
It is mentioned [here](https://github.com/kata-containers/tests/blob/6838b47bf4f82e6c70c7f032e40865bea01f6888/.ci/jenkins_job_build.sh#L132-L134) that, 

> We also run static checks on travis for x86 and ppc64le,

AFAIU, most of the CI is run on jenkins. Are there any checks/tests run on Travis which I might not be aware of as mentioned above?

After a small discussion with @Jakob-Naucke on this PR- [kata-containers#2736](https://github.com/kata-containers/kata-containers/pull/2736), it was discovered that even though enabled, the static-checks do not run on `s390x` and `aarch64` - https://github.com/kata-containers/tests/blob/6838b47bf4f82e6c70c7f032e40865bea01f6888/.ci/jenkins_job_build.sh#L130-L141

The reason being the condition used on L131, where the value of `METRICS_CI`  is never empty(either set to true or is false by default). After changing the condition to check against `METRICS_CI` being false and testing it locally by adding `ppc64le`, static checks run and the error being reported in the PR -  [kata-containers#2736](https://github.com/kata-containers/kata-containers/pull/2736) can be seen.

```
kata-check_ppc64le_test.go:50:40: cannot use false (untyped bool constant) as string value in struct literal (typecheck)
                {filepath.Join(sysModuleDir, "kvm"), false, "Y"},
                                                     ^
kata-check_ppc64le_test.go:51:43: cannot use false (untyped bool constant) as string value in struct literal (typecheck)
                {filepath.Join(sysModuleDir, "kvm_hv"), false, "Y"},
                                                        ^

```

**Comment by Amulyam24** [comment]
@Jakob-Naucke , @jongwu , can you'll confirm if I can go ahead and enable running static checks via a new PR on non x86?

**Comment by jongwu** [comment]
Hello @Amulyam24 -, I think it's a good point to do static checks and unit test on non-x86 as travis is gone.

**Comment by Jakob-Naucke** [comment]
@Amulyam24 thank you for raising this. Indeed, the tests on Travis were decommissioned before you or I joined the project I think, so this comment is quite outdated. These days, it should be `# We run static checks on GitHub Actions for x86_64, so run them on Jenkins for non-x86_64 only`. Feel free to go ahead with a PR. Please do check that tests like that of `kata-check` are really being run then, e.g. by not applying your changes from https://github.com/kata-containers/kata-containers/pull/2736.

**Comment by Amulyam24** [comment]
> Please do check that tests like that of `kata-check` are really being run then, e.g. by not applying your changes from [kata-containers/kata-containers#2736](https://github.com/kata-containers/kata-containers/pull/2736).

@Jakob-Naucke , do you mean running all the tests of `static-check` here?

**Comment by Jakob-Naucke** [comment]
> @Jakob-Naucke , do you mean running all the tests of `static-check` here?

Heh. I looked a bit into `static-check.sh` yesterday, and there were... some failures, and none that looked like they belonged to `kata-check`. The script is huge though and I have not yet had the opportunity to properly find my way around it. I am just saying we should run every test that might fail architecture-specifically.

**Comment by Amulyam24** [comment]
Yes, I'll go ahead and start with enabling static checks and UT on non-x66 for now and probably we can build and work from there if it makes sense?

**Comment by Jakob-Naucke** [comment]
> Yes, I'll go ahead and start with enabling static checks and UT on non-x66 for now and probably we can build and work from there if it makes sense?

Sure. I can also continue my investigations there (not sure about today though).
